tests/src/kernel/hipLaunchParm.cpp

```diff
@@ -23,75 +23,768 @@ THE SOFTWARE.
+}
+
+// Passing empty struct to a hipLaunchKernelGGL(),
+// check the size of 1Byte, set  result_d4 to true if condition met
+__global__ void hipLaunchKernelStructFunc4(
+                    hipLaunchKernelStruct_t4 hipLaunchKernelStruct_,
+                    bool* result_d4) {
+    int x = blockIdx.x * blockDim.x + threadIdx.x;
+
+    // set the result to true if the condition met
+    result_d4[x] =  (sizeof(hipLaunchKernelStruct_) == 1);
+}
+
+// Passing struct with pointer object to a hipLaunchKernelGGL()
+__global__ void hipLaunchKernelStructFunc5(
+                    hipLaunchKernelStruct_t5 hipLaunchKernelStruct_,
+                    bool* result_d5) {
+    int x = blockIdx.x * blockDim.x + threadIdx.x;
+
+    // set the result to true if the condition met
+    result_d5[x] =  ((hipLaunchKernelStruct_.c1 == 'c')
+                      && (*hipLaunchKernelStruct_.cp == 'p'));
+}
+
+// Passing struct which is aligned to 8Byte to a hipLaunchKernelGGL(),
+// set the result_d6 to true if condition met
+__global__ void hipLaunchKernelStructFunc6(
+                    hipLaunchKernelStruct_t6 hipLaunchKernelStruct_,
+                    bool* result_d6) {
+    int x = blockIdx.x * blockDim.x + threadIdx.x;
+
+    // set the result to true if the condition met
+    // get the address of the struct
+    // size_t(p)%8 will be 0 if aligned to 8Byte address space
+    int *p = (int*)(&hipLaunchKernelStruct_);
+    result_d6[x] =  ((hipLaunchKernelStruct_.c1 == 'c')
+                      && (hipLaunchKernelStruct_.si == 1)
+                      && ((size_t(p))%8 ==0));
+}
+
+// Passing struct which is aligned to 16Byte,
+// set the result_d7 to true if condition met
+__global__ void hipLaunchKernelStructFunc7(
+                    hipLaunchKernelStruct_t7 hipLaunchKernelStruct_,
+                    bool* result_d7) {
+    int x = blockIdx.x * blockDim.x + threadIdx.x;
+
+    // set the result to true if the condition met
+    // get the address of the struct
+    // size_t(p)%16 will be 0 if aligned to 16Byte address space
+    int *p = (int*)(&hipLaunchKernelStruct_);
+    result_d7[x] =  ((hipLaunchKernelStruct_.c1 == 'c')
+                      && (hipLaunchKernelStruct_.si == 1)
+                      && ((size_t(p))%16 ==0) );
+}
+
+// Passing struct which is packed & aligned to 4Byte,
+// set the result_d8 to true if condition met
+__global__ void hipLaunchKernelStructFunc8(
+                    hipLaunchKernelStruct_t8 hipLaunchKernelStruct_,
+                    bool* result_d8) {
+    int x = blockIdx.x * blockDim.x + threadIdx.x;
+
+    // set the result to true if the condition met
+    // get the address of the xth element, struct[x],
+    // size_t(p)%4 will be 0 if aligned to 4Byte address space
+    int *p = (int*)(&hipLaunchKernelStruct_);
+    result_d8[x] =  ((hipLaunchKernelStruct_.c1 == 'c')
+                      && (hipLaunchKernelStruct_.si == 1)
+                      && ((size_t(p))%4 ==0) );
```

Comment:
also need to check sizeof(hipLaunchKernelStruct_)==6

Reply:
i guess it's 4bytes not 6, sizeof(hipLaunchKernelStruct_)==4.
Done. Add the size validation condition to the kernel